deltachat-ios/Helper/MediaPicker.swift

```diff
@@ -75,28 +76,42 @@ class MediaPicker: NSObject, UINavigationControllerDelegate {
         navigationController?.present(audioRecorderNavController, animated: true)
     }
 
-    func showDocumentLibrary(selectFolder: Bool = false) {
+    func showFilesLibrary() {
+        let alert = UIAlertController(title: nil, message: String.localized("files_attach_hint"), preferredStyle: .safeActionSheet)
+        alert.addAction(UIAlertAction(title: String.localized("choose_from_files"), style: .default) { [weak self] _ in
+            self?.showDocumentLibrary()
+        })
+        alert.addAction(UIAlertAction(title: String.localized("choose_from_gallery"), style: .default) { [weak self] _ in
+            self?.showGallery(sendAsFile: true)
+        })
+        alert.addAction(UIAlertAction(title: String.localized("cancel"), style: .cancel))
+        navigationController?.present(alert, animated: true)
+    }
+
+    func showDocumentLibrary(selectBackupArchives: Bool = false) {
         let documentPicker: UIDocumentPickerViewController
-        if selectFolder {
+        if selectBackupArchives {
             documentPicker = .init(forOpeningContentTypes: [UTType.archive], asCopy: false)
         } else {
-            let types = [UTType.pdf, .text, .rtf, .spreadsheet, .vCard, .zip, .image, .data]
-            documentPicker = .init(forOpeningContentTypes: types, asCopy: true)
+            documentPicker = .init(forOpeningContentTypes: [UTType.pdf, .text, .rtf, .spreadsheet, .vCard, .zip, .image, .data], asCopy: true)
         }
         documentPicker.delegate = self
         documentPicker.allowsMultipleSelection = false
         documentPicker.modalPresentationStyle = .formSheet
         navigationController?.present(documentPicker, animated: true)
     }
 
-    func showGallery(allowCropping: Bool = false) {
+    func showGallery(allowCropping: Bool = false, sendAsFile: Bool = false) {
         // we have to use older UIImagePickerController as well as newer PHPickerViewController:
         // - only the older allows cropping and only the newer allows mutiple selection
         // - the newer results in weird errors on older OS, see discussion at https://github.com/deltachat/deltachat-ios/pull/2678
+        self.sendAsFile = sendAsFile
```

Comment:
hmmm mediapicker is stored on chatviewcontroller so setting this and never clearing it makes it set for the next attachment you add too? so:
- send image as file 
  - `onMediaSelected` called with sendAsFile true, as expected
- then send an image from the camera
  - now `onMediaSelected` is called with sendAsFile true (I think?) because `ChatViewController.mediaPicker.sendAsFile` is still true

Reply:
i added a commit

Reply:
> paths not starting with `showGallery()` do not end up in `onMediaSelected()`

Ah I did not realize this, feel free to ignore.